### ISSUE TITLE
dialog.transitionComponent prop

### ISSUE THREAD

**Issue by ShaneMasuda** [cmeeren/Feliz.MaterialUI#47]
What is the proper way to use the transitionComponent prop for dialogs? The Material UI documentation uses the forwardRef function as

const Transition = React.forwardRef(function Transition(props, ref) {
  return <Slide direction="up" ref={ref} {...props} />;
});

but the Feliz.React.forwardRef function returns a ReactElement rather than a ReactElementType. Is there an additional function call required to convert from ReactElement to ReactElementType, or is there a simpler way to go about this?

**Comment by Shmew** [comment]
Hi @ShaneMasuda, try unboxing and see if it works.

**Comment by ShaneMasuda** [comment]
@Shmew unboxing the return value of the forwardRef function into a ReactElementType? That doesn't seem to work for me, and neither does the following:

![Capture](https://user-images.githubusercontent.com/13651094/84823710-a3d7a380-afd3-11ea-8596-2f0c71775e94.PNG)

**Comment by Shmew** [comment]
I *think* this will work `unbox<ReactElementType>(System.Func<_,_,_>(fun props ref -> transition))`

@cmeeren, @zaid-ajaj thoughts on how APIs that need `ReactElementType` can be better handled?

**Comment by ShaneMasuda** [comment]
@Shmew so I tried that as

![Capture2](https://user-images.githubusercontent.com/13651094/84830711-c7541b80-afde-11ea-9674-b473849eef96.PNG)

but it seems the slide component still doesn't know about the ref. Namely, I'm getting the following error:

![Capture3](https://user-images.githubusercontent.com/13651094/84830731-d76bfb00-afde-11ea-8d40-bf94d3394536.PNG)

Am I perhaps not using the forwardRef function correctly? Thank you for responding so quickly btw!

**Comment by Shmew** [comment]
Yeah, you're using it correctly. I'm not really sure you even need it here, you could just try making the element itself and unboxing that, or `unbox<ReactElementType>(System.Func<_,_,_>(fun props ref -> transition(props,ref))`

**Comment by ShaneMasuda** [comment]
Unboxing the element itself yielded this:

![Capture5](https://user-images.githubusercontent.com/13651094/84832924-ff5d5d80-afe2-11ea-96d9-eaeb22342f9f.PNG)

and using the last line yielded this:

![Capture4](https://user-images.githubusercontent.com/13651094/84832950-0dab7980-afe3-11ea-8538-01af3711d9aa.PNG)

Is there any way you could try to get a simple example dialog up and running on your end? It would be greatly appreciated!

**Comment by Shmew** [comment]
I just spent some time trying to figure this out, I'm not sure how to fix it right now :(. I could have sworn I already battled this issue before.

**Comment by ShaneMasuda** [comment]
No worries! If any updates happen I'll be here. In the meantime I can work around it by making my own backdrop and making the dialog a child of the slide component.

**Comment by cmeeren** [comment]
@Zaid-Ajaj do you have any input here? `React.forwardRef` is from Feliz, I suppose?

**Comment by Zaid-Ajaj** [comment]
@cmeeren I have no idea, a reproducible example would be nice to see so that we can test what the problem is and compare with how the code looks like in native Mui

**Comment by marcpiechura** [comment]
Hey, 

I run into the same issue. 
@ShaneMasuda have you been able to resolve this one or are you still using your workaround?

thanks.

**Comment by ghost** [comment]
@marcpiechura @Shmew @Zaid-Ajaj coming back to this issue because I found a fix, I created a ReactElementType from scratch and combined the keys and values manually as custom props, which seems to do the trick:

![image](https://user-images.githubusercontent.com/13651094/131900121-783667d4-34e4-406f-bedc-754011dd9bee.png)

The main issue here is that React.forwardRef always seems to yield null for the ref. Please let me know if anything changes!
